Andrew Kushyk Person class

git-user-cpp opened the pull request:
**Changes:**

1. file **_.gitignore_** moved to **_NaiveSQL_** repository to enable its work in other IDEs.
2. Class **_Person_** was implemented _(and then re-written using List for storing fields)_.
3. Other commits were testing connection due to some problems with GitHub.

git-user-cpp committed:
.gitignore moved to a higher repository, skeleton of Person class was written
- file .gitignore moved to NAIVESQL repository (for working in other IDEs)
- skeleton of Person class was written (will be implemented soon)

git-user-cpp committed:
Class Person was implemented
- added fields and basic output function (fields should be re-writed by using List)

git-user-cpp committed:
Class Person was re-written
- fields logic was re-written (standart fields were replaced by list)
- function printPerson() was re-written due to using list for output

git-user-cpp committed:
In file Person.java
- List<String> fields: was changed to private due to security standarts

git-user-cpp committed:
test commit

git-user-cpp committed:
In file Person.java
- List<String> fields: was changed to private due to security standarts

VladimirPetrat commented on MainModule/src/main/java/models/Person.java:
```diff
@@ -0,0 +1,28 @@
+
+public class Person {
+
+    //list of Strings for holding data
```
Remove comments, add Javadoc on later versions. Same for all

VladimirPetrat commented on MainModule/src/main/java/models/Person.java:
```diff
@@ -0,0 +1,28 @@
+public class Person {
+
+    //list of Strings for holding data
+    private List<String> fields = new ArrayList<String>();
```
Use lazy initialization in constructor

VladimirPetrat commented on MainModule/src/main/java/models/Person.java:
```diff
@@ -0,0 +1,28 @@
+    //output function
+    public void printPerson(){
+
+        int i = 0;
```
Use iterator of Collections object, avoid using unnecessary primitives.

git-user-cpp committed:
In file Person.java
- regular comments were replaced by JavaDoc
- in constructor was implemented lazy initialization
- output implemented via using iterator

git-user-cpp committed:
File Person.java was renamed + getters and setters were implemented

VladimirPetrat commented on MainModule/src/main/java/models/PersonDTO.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.List;
+import java.util.ArrayList;
+
+public class PersonDTO {
```
Use Lombok @Data annotation to prevent b-plate code

VladimirPetrat commented on MainModule/src/main/java/models/PersonDTO.java:
```diff
@@ -0,0 +1,76 @@
+    public void printPerson() {
+        if (this.fields == null) {
+            this.fields = new ArrayList<String>();
+            this.fields.add(this.id);
```
Avoid using double logic in methods, separate to 2 methods

VladimirPetrat commented on MainModule/src/main/java/models/PersonDTO.java:
```diff
@@ -0,0 +1,76 @@
+    /**
+     * Fields
+     */
+    private String id, name, year;
```
What if client code will need to add additional field to class?

git-user-cpp committed:
In file PersonDTO.java
- implemented HashMap for storing fields
- implemented constructor for initializing fields
- implemented a getter for HashMap
- function printPerson() was re-written

git-user-cpp committed:
Changes:
In file StartClass.java:
    - output of object was re-implemented
In file PersonDTO.java:
    - method toString was implemented
    - unnececary methods were removed

git-user-cpp committed:
In file PersonDTO.java implemented a method returnPersonDTO()

git-user-cpp committed:
In file PersonDTO.java implemented a method replaceID()

git-user-cpp committed:
In file PersonDTO.java
- method operationReplace was implemented to make replacement easier
- method replaceName() was implemented
- method replaceYear() was implemented

git-user-cpp committed:
In file PersonDTO.java
- method returnId() was implemented
- method returnName() was implemented
- method returnYear() was implemented

git-user-cpp committed:
In file PersonDTO.java code was optimized:
- by adding a method returnField
- by adding a method replaceField

VladimirPetrat commented on MainModule/src/main/java/models/PersonDTO.java:
```diff
@@ -0,0 +1,40 @@
+        return fields.get(field);
+    }
+
+    private void operationReplace(String field, String newField){
```
Code format. Same for others

VladimirPetrat commented on MainModule/src/main/java/StartClass.java:
```diff
@@ -1,6 +1,20 @@
+        me.replaceField("year", "1999");
+        System.out.println(me.returnPersonDTO());
+
+        System.out.println("\nID = " + me.returnField("id") +
```
Use line separator method to guarantee cross platform traceability

VladimirPetrat commented on MainModule/src/main/java/models/PersonDTO.java:
```diff
@@ -0,0 +1,40 @@
+        this.fields.put("year", year);
+    }
+
+    public HashMap<String, String> returnPersonDTO(){
```
Remove, check @Data annotation

git-user-cpp committed:
In file PersonDTO.java:
- method addField was implemented
- method removeField was implemented

git-user-cpp committed:
@Data was removed

git-user-cpp committed:
In file StartClass.java used System.lineSeparator() instad of '/n'

git-user-cpp committed:
Visual changes due to code syle of the project

git-user-cpp committed:
Visual changes due to code syle of the project

git-user-cpp committed:
In file PersonDTO.java method replaceField was optimized
